os/test/src-jvm/PathTestsJvmOnly.scala

```diff
@@ -40,6 +44,18 @@ object PathTestsJvmOnly extends TestSuite {
             names.foreach(p => assert(!exists(twd / p)))
           }
         }
+        test("custom filesystem") { // native doesnt support custom fs yet
+          val path: java.nio.file.Path = java.nio.file.Paths.get("foo.jar");
+          val uri = new URI("jar", path.toUri().toString(), null);
+
+          val env = new HashMap[String, String]();
+          env.put("create", "true");
+
+          val fileSystem = FileSystems.newFileSystem(uri, env);
+          val p = os.root("/", fileSystem) / "test" / "dir"
+          assert(p.root == "/")
+          assert(p.fileSystem == fileSystem)
```

Comment:
Could we add a few more operations here? e.g. running `os.read`, `os.write`, `os.copy`, `os.move` on files with `foo.jar`.

Perhaps in that case it's worth moving this test suite into its own file